### DISCUSSION TITLE
[Chapter 12] 멀티 스레드로 이벤트를 처리하는 2가지 설정 방법

### DISCUSSION THREAD

**Discussion by coalong** [cowaine/spring-boot-for-msa#201] [Chapter 12. Spring Events] [answered]
P.720

> 이 장에서는 두 가지 방법을 사용하여 멀티 스레드로 이벤트를 처리하는 방법 을 설명한다. 
첫 번째는 이벤트 메시지를 전달하는 ApplicationEventMulticaster를 설정하는 방법이며, 
두 번째는 스프링 프레임워크에서 제공하는 `@Async` 애너테이션을 사용하는 방법이다.

예전에 프로젝트를 했을 당시 Spring Event 를 도입하여 회원이 회원가입시 쿠폰을 발급하는 기능을 구현해본 적이 있다.

그때는 스프링에서 제공하는 `@Async` 애너테이션을 사용했는데,
ApplicationEventMulticaster 를 설정하여 멀티 스레드로 이벤트를 구독하는 방법을 처음 알았다.

실제로 Spring Event 를 샤용해본 경험이 있다면 어떤 상황에서 Event 로 구독/게시 기능이 필요했는지?
이 두가지 비동기 설정 방법 중 어떤 방법을 사용하는지? 장단점은 없을까?

ApplicationEventMulticaster 설정과 `@EnableAsync` 설정을 둘 다 사용하면 우선순위?

**Comment by JoisFe** [comment, marked as answer]
Spring Event를 사용하고 있는데 클래스간 결합도를 낮출때 매우 유용했습니다.

해당 부분을 정말 이해하기 쉽게 설명한 영상이 있어 공유드립니다.
[참고 영상](https://www.youtube.com/watch?v=TJUIkLFpgGo)
